internal/internal_workflow_testsuite_test.go

```diff
@@ -1274,6 +1274,228 @@ func (s *WorkflowTestSuiteUnitTest) Test_GetVersion() {
+		if v == DefaultVersion {
+			f = ExecuteActivity(ctx, oldActivity, "ols_msg")
+		} else {
+			f = ExecuteActivity(ctx, newActivity, "new_msg")
+		}
+		err := f.Get(ctx, nil) // wait for result
+		if err != nil {
+			return err
+		}
+
+		// test searchable change version
+		wfInfo := GetWorkflowInfo(ctx)
+		s.NotNil(wfInfo.SearchAttributes)
+		changeVersionsBytes, ok := wfInfo.SearchAttributes.IndexedFields[CadenceChangeVersion]
+		s.True(ok)
+		var changeVersions []string
+		err = json.Unmarshal(changeVersionsBytes, &changeVersions)
+		s.NoError(err)
+		s.Equal(1, len(changeVersions))
+		s.Equal("test_change_id-1", changeVersions[0])
+
+		return err
+	}
+
+	env := s.NewTestWorkflowEnvironment()
+	env.RegisterWorkflow(workflowFn)
+	env.RegisterActivity(oldActivity)
+	env.RegisterActivity(newActivity)
+	env.OnActivity(newActivity, mock.Anything, "new_msg").Return("hello new_mock_msg", nil).Once()
+	env.ExecuteWorkflow(workflowFn)
+
+	s.True(env.IsWorkflowCompleted())
+	s.Nil(env.GetWorkflowError())
+	env.AssertExpectations(s.T())
+}
+
+func (s *WorkflowTestSuiteUnitTest) Test_MockGetVersion_ExecuteWithVersion() {
+	oldActivity := func(ctx context.Context, msg string) (string, error) {
+		return "hello" + "_" + msg, nil
+	}
+	newActivity := func(ctx context.Context, msg string) (string, error) {
+		return "hello" + "_" + msg, nil
+	}
+	workflowFn := func(ctx Context) (string, error) {
+		ctx = WithActivityOptions(ctx, s.activityOptions)
+		var f Future
+		v1 := GetVersion(ctx, "change_1", DefaultVersion, 2, ExecuteWithVersion(1))
+		if v1 == DefaultVersion {
+			f = ExecuteActivity(ctx, oldActivity, "old1")
+		} else {
+			f = ExecuteActivity(ctx, newActivity, "new1")
+		}
+		var ret1 string
+		err := f.Get(ctx, &ret1) // wait for result
+		if err != nil {
+			return "", err
+		}
+
+		v2 := GetVersion(ctx, "change_2", DefaultVersion, 2)
+		if v2 == DefaultVersion {
+			f = ExecuteActivity(ctx, oldActivity, "old2")
+		} else {
+			f = ExecuteActivity(ctx, newActivity, "new2")
+		}
+		var ret2 string
+		err = f.Get(ctx, &ret2) // wait for result
+		if err != nil {
+			return "", err
+		}
+
+		// test searchable change version
+		wfInfo := GetWorkflowInfo(ctx)
+		s.NotNil(wfInfo.SearchAttributes)
+		changeVersionsBytes, ok := wfInfo.SearchAttributes.IndexedFields[CadenceChangeVersion]
+		s.True(ok)
+		var changeVersions []string
+		err = json.Unmarshal(changeVersionsBytes, &changeVersions)
+		s.NoError(err)
+		s.Equal(2, len(changeVersions))
+		s.Equal("change_2-2", changeVersions[0])
+		s.Equal("change_1--1", changeVersions[1])
+
+		return ret1 + ret2, err
+	}
+
+	env := s.NewTestWorkflowEnvironment()
+	env.RegisterWorkflow(workflowFn)
+	env.RegisterActivity(oldActivity)
+	env.RegisterActivity(newActivity)
+
+	env.OnGetVersion("change_1", DefaultVersion, 2).Return(func(string, Version, Version) Version {
+		return DefaultVersion
+	})
+	env.OnGetVersion(mock.Anything, DefaultVersion, 2).Return(Version(2))
```

Comment:
```suggestion
	env.OnGetVersion("change_1", DefaultVersion, 2).Return(DefaultVersion)
	env.OnGetVersion("change_2", DefaultVersion, 2).Return(Version(2))
```
since it's weirdly inconsistent and loose otherwise (also below)